### ISSUE TITLE
Rectangular input shape results in square output shape

### ISSUE THREAD

**Issue by Fritskee** [qubvel/segmentation_models.pytorch#475]
Thanks for the implementation of SMP, I have been playing with this project and noticed the following:
- My dataset consists of images with shape 512x288
- My output results are of the shape 512x512, this means I manually need to trim a section of each image, in order to obtain the correct result. Besides that, it also makes me think that there are quite some calculations happening (factor 1.7) that should not be happening.

Does anybody know how this phenomenon arises? I expected it to be in the dataloader, or either in the upsampling, but in either case I cannot really what I should change, in order to get the 512x288 output, instead of the default 512x512

**Comment by qubvel** [comment]
Could you provide a minimal example to reproduce, e.g.

```python
model = smp.Unet()
out = model(tensor.ones([1, 3, 512, 288]))
out.shape
```

**Comment by Fritskee** [comment]
`
model = smp.Unet(encoder_name="efficientnet-b0", encoder_weights="imagenet", classes=["0", "1", "2"], activation="softmax2d", in_channels=3)


image, gt_mask = test_dataset[n]

gt_mask = gt_mask.squeeze()
gt_mask = onehot_to_class_index(gt_mask)

x_tensor = torch.from_numpy(image).to(DEVICE).unsqueeze(0)
pr_mask = best_model.predict(x_tensor)
pr_mask = (pr_mask.squeeze().cpu().numpy().round())
pr_mask = onehot_to_class_index_preds(pr_mask)
`
after this, both the ground truth mask (gt_mask) and the predicted mask (pr_mask) need to be resized. However, originally the image and mask have the shape (512, 288, 3)

**Comment by Fritskee** [comment]
@qubvel I use `preprocessing_fn = smp.encoders.get_preprocessing_fn(ENCODER, ENCODER_WEIGHTS)`
so if any augmentations happen, it is in that function. I personally did not program any augmentations to happen.

**Comment by Fritskee** [comment]
Solved, it was a dataloader issue
